isis/src/base/objs/PvlTranslationTable/PvlTranslationTable.cpp

```diff
@@ -365,97 +372,149 @@ namespace Isis {
    * Returns the input default value from the translation table corresponding
    * to the output name argument.
    *
-   * @param nName The output name to be used to search the translation table.
+   * @param translationGroupName The output name to be used to search the translation table.
    *
    * @return QString The input default value
    *
-   * @throws iException::Programmer
+   * @throws IException::Programmer
    */
-  QString PvlTranslationTable::InputDefault(const QString nName) const {
+  QString PvlTranslationTable::InputDefault(const QString translationGroupName) const {
 
-    if(!p_trnsTbl.hasGroup(nName)) {
-      QString msg = "Unable to find translation group [" +
-                   nName + "] in file [" + p_trnsTbl.fileName() + "]";
-      throw IException(IException::Programmer, msg, _FILEINFO_);
-    }
+    const PvlGroup &translationGroup = findTranslationGroup(translationGroupName);
 
-    PvlGroup tgrp = p_trnsTbl.findGroup(nName);
-    if(tgrp.hasKeyword("InputDefault")) return tgrp["InputDefault"];
+    if (translationGroup.hasKeyword("InputDefault")) return translationGroup["InputDefault"];
```

Comment:
Should this use `hasInputDefault()` ?

Reply:
maybe not.  hasInputDefault(transGrpName) uses the group name to find the group first before determining whether the group has this keyword. 
in the translation method we have already searched and found the group and plan to use it elsewhere. 
I'm not sure why the hasInputDefault() method exists. It appears to only be used in XmlToPvlTranslationManager. And appears to be an inefficient design, but this is older code and out of the scope of this bug fix.

Reply:
:+1: Thanks for explaining.